### ISSUE TITLE
borg mount / slow / 100% cpu usage / mac

### ISSUE THREAD

**Issue by devrando** [borgbackup/borg#1398]
I created a local backup with borg 1.0.6 on Mac El Capitan of a 6GB Raw File.
After mounting the repo locally (borg mount using OSXFuse 3.3.3)I tried to copy the file.
Perfomance is really bad:
```
rsync  stdin --size-only --progress /Users/myuser/
Password:
stdin
    31227904   0%   74.86kB/s   29:17:01
```

"borg mount" process consumes 100%cpu (Core i7)

**Comment by enkore** [comment]
There is a known performance issue in the FUSE implementation in the 1.0 series (#966).

If borg-extract shows adequate performance then this is a duplicate of #966

**Comment by ThomasWaldmann** [comment]
Also, in general, the fuse mount is rather for convenience than performance.
If you know you will deal with huge files or with lots of files, borg extract is usually quicker.

**Comment by devrando** [comment]
I'll try it with extract, but i have **no** performance issues with my other productive backup(borg 1.0.3/linux debian jessie64).
Using borg mount is very convenient if you want to startup a backup VM with `qemu -hda /my/rawfile.img -snapshot`

Currently rsync calculates **24h for 6GB file**
(Thanks for this fast feedback)

**Comment by enkore** [comment]
When I `strace´ an `rsync` then I see that it often does a 32k read followed by one or more 4 byte (!) or 8 byte reads, e.g.

```
pid   653] read(0, "..."..., 32768) = 32768
[pid   653] read(0, "\366k3\24if\260#", 8) = 8
[pid   653] read(0, "..."..., 32768) = 32768
```

Let's suppose that 2 out of 3 reads are not 32k but 4 or 8 bytes (which seems to be approx the case here) then the average read size is only 10k. Less if these short reads happen more often. Thus we arrive at an average slowdown of 2 MB / 10 kB (or less) = 200 or more. If borg-extract would manage, say, 50 MB/s, then this will lead to only 250 kB/s through FUSE on 1.0.x



I am not sure how qemu handles IO to disk files, but if it pretty much reads them on a page-by-page basis (4 kB) then it's probably bad performance too.

@ThomasWaldmann : Backport #965 to 1.0-maint for 1.0.7?

**Comment by ThomasWaldmann** [comment]
Yes, seems simple enough.

**Comment by devrando** [comment]
Just checked current master.
Much better:

```
bash-3.2# rsync  -r stdin --size-only --progress /Users/myuser/
building file list ... 
1 file to consider
stdin
   604798976   7%    4.28MB/s    0:27:48
```

`qemu-system-x86_64 -hda stdin  -snapshot -m 1024`

Works also as expected.
Fantastic support, waiting for 1.0.7

**Comment by PlasmaPower** [comment]
I'm running tests on a cherry pick right now, PR should be up soon.

**Comment by PlasmaPower** [comment]
With #1401 merged this will be coming in Borg 1.0.7, the next release.

**Comment by ThomasWaldmann** [comment]
Fixed by #1401.
